src/embedded.js

```diff
@@ -606,11 +609,18 @@
                 window.addEventListener('scroll', this.fixIframe);
             }
 
+            // Close the iframe if page fails to initialize within 15 seconds
+            var iframeHealthCheck = setTimeout(function() {
+                HelloSign.reportError('Signer page failed to initialize within 15 seconds.', document.location.href);
+                HelloSign.close();
+            }, HelloSign.healthCheckInterval);
```

Comment:
I may be wrong about the scoping -- but seems like it might have potential for weird bugginess to not be explicit about the scope?